lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -1742,13 +1742,27 @@ checkIndividualConformance(NormalProtocolConformance *conformance,
     return conformance;
   }
 
-  // If the protocol requires a class, non-classes are a non-starter.
-  if (Proto->requiresClass() && !canT->getClassOrBoundGenericClass()) {
-    C.Diags.diagnose(ComplainLoc,
-                     diag::non_class_cannot_conform_to_class_protocol, T,
-                     ProtoType);
-    conformance->setInvalid();
-    return conformance;
+  if (canT->getClassOrBoundGenericClass()) {
```

Comment:
Can you put this next to the existing `ObjectiveCBridgeable` special case in ConformanceChecker::checkConformance()?